### PR TITLE
fix bug in leaving as the last owner

### PR DESCRIPTION
**Pull request by xgess**
a team is not valid without an owner, so the last owner may not leave it. in JS land, we don't bother letting a user make this attempt and instead show a helpful explanation. 

previously, we were showing this message only if you were the very last remaining user in the team. this is a sufficient but not necessary condition for being the final owner, for example if there is another non-owner user in the team. in this case we should also be preventing the request instead of letting the it go through and black-bar-ing.

### CODE REVIEW

**Comment by xgess** [comment]
@keybase/react-hackers  ^

File: shared/constants/teams.js [hunk 1]
```diff
@@ -265,6 +265,12 @@ const hasChannelInfos = (state: TypedState, teamname: Types.Teamname): boolean =
 const getTeamMemberCount = (state: TypedState, teamname: Types.Teamname): number =>
   state.teams.getIn(['teammembercounts', teamname], 0)
 
+const isLastOwner = (state: TypedState, teamname: Types.Teamname): boolean => {
+  const allTeamMembers = state.teams.teamNameToMembers.getIn([teamname], I.Map())
```
Comment (chrisnojima) [inline shared/constants/teams.js]: can be
`state.teams.teamNameToMembers.get(teamname, I.Map())`

File: shared/constants/teams.js [hunk 2]
```diff
@@ -265,6 +265,12 @@ const hasChannelInfos = (state: TypedState, teamname: Types.Teamname): boolean =
...
+  const countOfOwners = allTeamMembers.count(m => m.type === 'owner')
```
Comment (chrisnojima) [inline shared/constants/teams.js]: not a big deal but if you check your own role you can opt out of looking at this entire list of members

Comment (chrisnojima) [inline shared/constants/teams.js]: also not a big deal but you could bail earlier than going through the whole list

File: shared/constants/teams.js (hunk 1 above)
Comment (xgess) [reply shared/constants/teams.js]: done

File: shared/constants/teams.js (hunk 2 above)
Comment (xgess) [reply shared/constants/teams.js]: done

Comment (xgess) [reply shared/constants/teams.js]: done

File: shared/constants/teams.js [hunk 3]
```diff
@@ -265,6 +265,21 @@ const hasChannelInfos = (state: TypedState, teamname: Types.Teamname): boolean =
 const getTeamMemberCount = (state: TypedState, teamname: Types.Teamname): number =>
   state.teams.getIn(['teammembercounts', teamname], 0)
 
+const isLastOwner = (state: TypedState, teamname: Types.Teamname): boolean =>
+  isOwner(getRole(state, teamname)) && !isMultiOwnerTeam(state, teamname)
+
+const isMultiOwnerTeam = (state: TypedState, teamname: Types.Teamname): boolean => {
+  let countOfOwners = 0
+  const allTeamMembers = state.teams.teamNameToMembers.get(teamname, I.Map())
+  const moreThanOneOwner = allTeamMembers.valueSeq().some(tm => {
```
Comment (chrisnojima) [inline shared/constants/teams.js]: don't need valueseq here, Map has a some() function

Comment (xgess) [reply shared/constants/teams.js]: done

**Pull request merged**